File: java/graphql-client/src/main/java/com/sdl/web/pca/client/response/IGraphQLResponse.java
package com.sdl.web.pca.client.response;

import java.util.List;

/**
 * This interface represent the Graphql response.
 * @author
 * @version 1.0.0
 */
public interface IGraphQLResponse {
    Object Data = null;
    List<GraphQLError> Errors = null;

    void setData(Object data);
    Object getData();

    void setErrors(List<GraphQLError> errors);
    List<GraphQLError> getErrors();
}

File: net/src/Sdl.Tridion.Api.Client/GraphQLClient/Response/IGraphQLResponse.cs
using System.Collections.Generic;
using Sdl.Tridion.Api.Http.Client;

namespace Sdl.Tridion.Api.GraphQL.Client.Response
{
    /// <summary>
    /// Represents the GraphQL respinse
    /// </summary>
    public interface IGraphQLResponse
    {
        /// <summary>
        /// GraphQL Data
        /// </summary>
        dynamic Data { get; set; }

        /// <summary>
        /// GraphQL Errors
        /// </summary>
        List<GraphQLError> Errors { get; set; }

        /// <summary>
        /// Response headers
        /// </summary>
        HttpHeaders Headers { get; set; }
    }
}
